pkg/reconciler/configuration/resources/revision.go

```diff
@@ -87,6 +87,11 @@ func updateRevisionAnnotations(rev, config metav1.Object) {
 		annotations[serving.CreatorAnnotation] = c
 	}
 
+	if v, ok := cans[serving.RoutesAnnotationKey]; ok {
+		annotations[serving.RoutesAnnotationKey] = v
+		rev.SetRoutingState(v1.RoutingStateActive, clock.RealClock{})
```

Comment:
I wonder if it's better as a free floating function and keep the wider interface?

Reply:
I feel like keeping this next to the creator annotation (or any annotations we're pulling from the config) makes more sense than splitting this out. The interface being objectmeta vs revision doesn't seem like a big deal.